keras/layers/pooling.py

```diff
@@ -424,7 +444,21 @@ class GlobalAveragePooling1D(_GlobalPooling1D):
     '''
 
     def call(self, x, mask=None):
-        return K.mean(x, axis=1)
+        if mask is None:
+            return K.mean(x, axis=1)
+        else:
+            mask = K.cast(mask, K.floatx())
+            # Expand count so we can do a broadcast division
+            count = K.sum(mask, axis=1)
+            count = K.expand_dims(count, dim=1)
+            if K.backend() == 'tensorflow':  # tensorflow shape command dosn't work
```

Comment:
IMHO `int_shape` should not be a documented part of the backend then. And begin with `__int_shape` to denote that it should be treated as private. But that is a separate issue. 

How would I make this work for symbolic shapes when `reapeat_elements` didn't work with tensorflow's output from `shape` ? I have no problem making necessary changes when time is available, but I have no intention on debugging large parts of the Keras backend with Theano and Tensorflow to root out behavior inconsistencies.